File: ios/mono_kit/Sources/mono_kit/MonoKitPlugin.m
#import "MonoKitPlugin.h"
#import <mono_kit/mono_kit-Swift.h>

@implementation MonoKitPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftMonoKitPlugin registerWithRegistrar:registrar];
}
@end

File: macos/Classes/MonoKitPlugin.swift
import Cocoa
import FlutterMacOS

public class MonoKitPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "mono_kit", binaryMessenger: registrar.messenger)
    let instance = MonoKitPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
